### PR TITLE
fixes for depth_multiplier in SeparableConv2d and DepthwiseConv2d

### PR DESCRIPTION
**Pull request by dgburnette**
A# Description

The model python attached (model_asic.py) demonstrated that the shape printed by hls4ml (e.g. [None,32,32,8]) did not agree with the python model.summary() (e.g. [None,32,32,40)).
It turns out to be an incomplete handling of depth_multiplier in the SeparableConv2d and DepthwiseConv2d implementations.

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [x ] Bug fix (non-breaking change that fixes an issue)
- [ ] Documentation update
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] A new research paper code implementation
- [ ] Other (Specify)

## Tests

import os
import shutil
import sys
import hls4ml
import yaml
import pickle
import argparse
import numpy as np
import tensorflow as tf
import json
from tensorflow import keras
from tensorflow.keras.models import model_from_json
from tensorflow.keras import regularizers
from tensorflow.keras.constraints import max_norm
from qkeras import *
from sklearn.metrics import accuracy_score
 
# Usage:
#    python3 model.py
#      will build the model, perform analysis, generate HLS C++ and then synthesize it
 
## Function to create depthwise conv2d Neural Network model. For depthwiseconv2d, the kernel size will always be (1,1)
def create_model():
    model = tf.keras.Sequential()
    model.add(DepthwiseConv2D(kernel_size=(1,1), padding='valid', depth_multiplier=5, strides=1, dilation_rate=1, use_bias=True, input_shape=(32,32,8)))
    #Compiling the model
    model.compile(loss='sparse_categorical_crossentropy',
             optimizer='adam',
             metrics=['accuracy'])
    return model
 
## Function to save model architecture, weights, and configuration
def save_model(model, name=None):
    if name is None:
        name = model.name
    model.save(name + '.h5')
    model.save_weights(name + '_weights.h5')
    with open(name + '.json', 'w') as outfile:
        outfile.write(model.to_json())
    return
 
if __name__ == '__main__':
 
    print("============================================================================================")
    print("HLS4ML Version: ",hls4ml.__version__)
 
    parser = argparse.ArgumentParser(description='Configure and convert the model for Catapult HLS')
    parser.add_argument('--synth', action='store_true', help='Specify whether to perform Catapult build and synthesis')
 
    args = parser.parse_args()
 
    # Define output files/location:
    model_name = 'depthwiseconv2d'
    proj_name = 'myproject'
    out_dir = 'my-Catapult-test_asic'
 
    print("")
    print("Settings:")
    print('  model_name = '+model_name)
    print('  proj_name  = '+proj_name)
    print('  out_dir    = '+out_dir)
    print("")
 
    ## Create the model
    print("============================================================================================")
    print("Creating the model")
    model = create_model()
    model.summary()
 
    # Save model
    save_model(model, model_name)
 
    print("============================================================================================")
    print("Configuring HLS4ML")
    config_ccs = {}
    config_ccs['Backend'] = 'Catapult'
    config_ccs['ClockPeriod'] = 10
    config_ccs['HLSConfig'] = {'Model': {'Precision': 'ac_fixed<16,8>', 'ReuseFactor': 1, 'Strategy': 'Latency'}}
    config_ccs['IOType'] = 'io_stream'
    config_ccs['KerasH5'] = model_name+'_weights.h5'
    config_ccs['KerasJson'] = model_name+'.json'
    config_ccs['ProjectName'] = proj_name
    config_ccs['ASICLibs'] = 'saed32rvt_tt0p78v125c_beh'
    config_ccs['FIFO'] = 'hls4ml_lib.mgc_pipe_mem'
    config_ccs['OutputDir'] = out_dir
 
    # Create .yml file
    print("============================================================================================")
    print('Writing YAML config file: '+proj_name+'_config.yml')
    with open(proj_name+'_config.yml', 'w') as yaml_file:
        yaml.dump(config_ccs, yaml_file, explicit_start=False, default_flow_style=False)
 
    # Note: after this point the command-line hls4ml can be used to generate the C++ design and Catapult script
    #  shell%> hls4ml convert -c <project>_config.yml
    #  shell%> cd <outputdir>
    #  shell%> catapult -product ultra -file build_prj.tcl
 
    print("============================================================================================")
    print("HLS4ML converting model to HLS C++")
    hls_model_ccs = hls4ml.converters.keras_to_hls(config_ccs)
 
    print("============================================================================================")
    print("Compiling HLS C++ model")
    hls_model_ccs.compile()
 
    if args.synth:
        print("============================================================================================")
        print("Synthesizing HLS C++ model using Catapult")
        hls_model_ccs.build(csim=True, synth=True, cosim=False, validation=True, vsynth=False, sw_opt=True)
        # hls_model_ccs.build()
    else:
        print("============================================================================================")
        print("Skipping HLS - To run Catapult directly:")
        print('cd ' + out_dir + '; catapult -file build_prj.tcl')

**Test Configuration**:

## Checklist

- [x ] I have read the [guidelines for contributing](https://github.com/fastmachinelearning/hls4ml/blob/main/CONTRIBUTING.md).
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have installed and run `pre-commit` on the files I edited or added.
- [ ] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: hls4ml/model/layers.py [hunk 1]
```diff
@@ -912,34 +935,14 @@ def initialize(self):
 
 
 class Resize(Layer):
-    _expected_attributes = [
-        Attribute('in_height'),
-        Attribute('in_width'),
-        Attribute('out_height'),
-        Attribute('out_width'),
```
Comment (jmitrevs) [inline hls4ml/model/layers.py:919]: I don't understand why this is removed.

**Comment by jmitrevs** [comment]
I think we want to handle this in a different way. We want to split seperable into depthwise followed by pointwise. I will soon make a PR trying to do that. I'll try to add depth multiplier support there.
